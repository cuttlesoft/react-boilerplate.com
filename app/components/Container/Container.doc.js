import { describe, PropTypes } from 'react-desc'

export const doc = Container => {
  const DocumentedContainer = describe(Container)
    .description('A Container')
    .usage(
      `import { Container } from 'components/Container';
<Container />`,
    )

  DocumentedContainer.propTypes = {
    id: PropTypes.string
      .description('The DOM id attribute value to use for the element.')
      .defaultValue(false),
  }

  return DocumentedContainer
}
