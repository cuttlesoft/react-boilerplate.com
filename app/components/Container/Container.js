import React from 'react'
import PropTypes from 'prop-types'

// Components
import { Box } from '../Box'

/**
 * Container
 */
const Container = ({ children }) => (
  <Box margin="auto" pad="large">
    {children}
  </Box>
)

Container.propTypes = {
  children: PropTypes.any,
}

export default Container
