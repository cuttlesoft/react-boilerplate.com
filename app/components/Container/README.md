<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## Container
A Container

## Usage

```javascript
import { Container } from 'components/Container';
<Container />
```

## Properties

**id**

The DOM id attribute value to use for the element.

```
string
```
  