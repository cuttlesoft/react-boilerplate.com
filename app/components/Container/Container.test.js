import React from 'react'
import { render } from '@testing-library/react'

import Container from './Container'

const renderComponent = (props = {}) => render(<Container {...props} />)

/**
 *
 * Tests for Container
 *
 */
describe('Container', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
