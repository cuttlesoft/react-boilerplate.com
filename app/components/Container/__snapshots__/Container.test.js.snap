// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 ewTzDI"
  />
</div>
`;
